summaries/Evolutionary-testing-of-classes-summary.md

```diff
@@ -0,0 +1,81 @@
+
+### Unit testing of classes
+Unit testing of classes considers that each class being tested consists of constructor(s), method invocation within sertain set of input parameters
+
+### Evolutionary testing
+Evolutionary testing requires certain steps to be implemented in the algorithm including definition of parameters, maximum execution time, targets, automatically generated tests.
+The algorithms resuts in gererating test cases that are properly defined for at lest one target. 
+
+### Chromosomes
+Chromosomes includes number of input values to be used as parameters of methods and constructors that belong to a ceartain class.
+The specification of chromosomes thus must include the specification of the methods and the proper sequence of parameters.
+
+### Mutation operators
+By randomly selectiong mutation operators, chromosomes are being changed by the generic algorithm.
+#### Mutation of input value
+Randomly generated value replaces another with the same type.
+#### Constructor change
+One of the constructors is randomly replaced.
+#### Insertion of method invocation
+New methods are generated with generated input values
+#### Removal of method invocation
+Randomly choosen methods are being removed.
+#### One-point crossover
+During a union of chromosomes at a randomly selected line, the confliction variables are being renamed, constructors are added or removed, methods are added or removed.
+
+### Input generators
+Input generators are defined in the declaration of method signatures. 
+Default input generator are use if only type names are declared.
+The following types for generated rules are applied:
+- Integer and real numbers (only in the interval from 0 to 100)
+- Booleans (50% probability for true, and 50% for false)
+- Strings (a-zA-Z0-9) 
+
+##The Tool ETOC
+eToc tool(evolutionary Testing of classes) gives realisation of generic algorith for test case generation.
+It consists of the following components:
+ 
+###Branch instrumentor
+Branch instrumentor generates instrumented code where each control flow branch is individually described and traced in the process of execution. 
+
+### Chromosome former
+Produces new chromosomes and modifies existing chromosomes according to method signatures. 
+
+### Test case generator 
+Withinh the inner loop of the generic algorithm every test case identified by a chromosome is implemented.
+
+#### Test case executor
+After execution of test cases, a summary containing information of successfully passed test cases, as well as failed ones and not proper assertions, is being generated.
+
+## Experimental Results
+Experimental results have been generated by using eToc tool.
+The unit testing have been applied to classes existing in the standard Java library.
+
+###The procedure
+The Classes Under Test have been implemented using *Branch instrumentor*.
+After that *Test case generator* generates Junit test cases for each Class Under Test. 
```

Comment:
Ok. Should I change it in that case? I think I should follow the paper but if you both insist then I will change it